microceph/cmd/microcephd/main.go

```diff
@@ -63,33 +62,49 @@ func (c *cmdDaemon) Command() *cobra.Command {
 }
 
 func (c *cmdDaemon) Run(cmd *cobra.Command, args []string) error {
-	m, err := microcluster.App(microcluster.Args{StateDir: c.flagStateDir, Verbose: c.global.flagLogVerbose, Debug: c.global.flagLogDebug})
+	m, err := microcluster.App(microcluster.Args{StateDir: c.flagStateDir})
 	if err != nil {
 		return err
 	}
 
-	h := &config.Hooks{}
-	h.PostBootstrap = func(s *state.State, initConfig map[string]string) error {
+	h := &state.Hooks{}
+	h.PostBootstrap = func(ctx context.Context, s state.State, initConfig map[string]string) error {
 		data := common.BootstrapConfig{}
 		interf := interfaces.CephState{State: s}
 		common.DecodeBootstrapConfig(initConfig, &data)
-		return ceph.Bootstrap(interf, data)
+		return ceph.Bootstrap(ctx, interf, data)
 	}
 
-	h.PostJoin = func(s *state.State, initConfig map[string]string) error {
+	h.PostJoin = func(ctx context.Context, s state.State, initConfig map[string]string) error {
 		interf := interfaces.CephState{State: s}
-		return ceph.Join(interf)
+		return ceph.Join(ctx, interf)
 	}
 
-	h.OnStart = func(s *state.State) error {
+	h.OnStart = func(ctx context.Context, s state.State) error {
 		interf := interfaces.CephState{State: s}
-		return ceph.Start(interf)
+		return ceph.Start(ctx, interf)
 	}
 
 	h.PreRemove = ceph.PreRemove(m)
 
-	m.AddServers(api.Servers)
-	return m.Start(context.Background(), database.SchemaExtensions, nil, h)
+	daemonArgs := microcluster.DaemonArgs{
+		// Microcluster requires an explicit version to be supplied to the daemon.
+		// This will be readable alongside other server information over the `GET /core/1.0` endpoint.
+		// MicroCeph should define a project version here that can be used by consumers of the MicroCeph API
+		// to ensure that MicroCeph is installed within a supported series of revisions.
+		// In particular, a semantic version would be very useful here to distinguish between breaking changes
+		// and non-breaking bugfixes to a supported series.
+		Version: "UNKNOWN", // FIXME: Add an explicit version to MicroCeph.
```

Comment:
Suggest to set this to the full MicroCeph version, and use a build flag as done for the CLI here:
https://github.com/canonical/microceph/blob/36f71d77006388164e312a2a5d5c4ba699a140a2/snap/snapcraft.yaml#L320

Reply:
I've added this as the version string now. 

There was an unused `version` package which now stores this value both for the client and the daemon.